core/lib/instrument.c

```diff
@@ -1791,7 +1791,7 @@ create_and_initialize_module_data(app_pc start, app_pc end, app_pc entry_point,
             copy->segments[i].end = os_segments[i].end;
             copy->segments[i].prot = os_segments[i].prot;
         }
-    } else
+    } else if (segments != NULL)
```

Comment:
> Probably it's only required to keep the compiler happy. 

There should be a comment saying this: as is, the code is misleading b/c it looks like segments can legitimately be NULL, which makes the code look buggy by only checking that where it does: we'll allocate memory and not fill it, or try to allocate 0 bytes.